packages/util/src/clone-element-with-ref.js

```diff
@@ -0,0 +1,17 @@
+import { cloneElement } from 'react'
+
+import { isFunction, isRef } from './index.js'
+
+export function cloneElementWithRef(element, outerRef, props) {
+  const { ref } = element
+
+  return cloneElement(element, {
+    ...props,
+    ref: node => {
```

Comment:
Would you please help me understand the need for the different cases here?

Reply:
sure can. 

`if (isRef(outerRef)) outerRef.current = node` just checks that the new ref we're appending exists

`if (isFunction(ref)) ref(node)` supports the older ref style
`else if (isRef(ref)) ref.current = node` supports the newer refs from `createRef`